multinet/multinet/uploaders/csv.py

```diff
@@ -64,25 +65,25 @@ def upload(workspace, table):
     """
     app.logger.info('Bulk Loading')
 
+    # Read the request body into CSV format
     body = request.data.decode('utf8')
+    rows = list(csv.DictReader(StringIO(body)))
 
-    rows = csv.DictReader(StringIO(body))
+    # Set the collection, paying attention to whether the data contains
+    # _from/_to fields.  coll = None
     workspace = db.db(workspace)
+    if workspace.has_collection(table):
```

Comment:
Should perform validation before making the collection.

Reply:
See a5332ae.